fenixedu-ist-giaf-invoices/src/main/java/pt/ist/fenixedu/giaf/invoices/ErrorLogConsumer.java

```diff
@@ -0,0 +1,22 @@
+package pt.ist.fenixedu.giaf.invoices;
+
+@FunctionalInterface
+public interface ErrorLogConsumer {
+
+    void accept(final String oid,
```

Comment:
The same consumer is being used in different context withing the synchronisation process.
In each context we have different information available, so we don't always have the same objects available.
While this API is a bit overwhelming it allows a final report to be complete and comprehensible gathering all relevant information that will be presented to the end users.